AnimationGroup.lua

```diff
@@ -60,28 +60,49 @@ function CreateFrame(...)
     return frame
 end
 
-
 local function OnUpdate(self, elapsed)
     if self.paused then
         return
     end
-
-    self.time = self.time + (self.group.reverse and -elapsed or elapsed)
+    self.delaying = true
+
+    if not self.group.reverse and self.startdelay and self.startdelayTime < self.startdelay then
+        self.startdelayTime = self.startdelayTime + elapsed
+    elseif self.group.reverse and self.enddelay and self.enddelayTime < self.enddelay then
+        self.enddelayTime = self.enddelayTime + elapsed
+    elseif (not self.group.reverse and self.time < self.duration) or (self.group.reverse and self.time > 0) then
+        self.time = self.time + (self.group.reverse and -elapsed or elapsed)
+        self.delaying = false
+        self.progress = self.time / self.duration
+        self.progress = self.progress < 0 and 0 or self.progress
+        self.progress = self.progress > 1 and 1 or self.progress
+        self.smoothProgress = self.smoothing_func(self.progress).y
+    end
 
     if self.time > self.duration or (self.group.reverse and self.time < 0) then
-        AG:Stop(self)
-        AG:Fire(self.group, self, 'Finished')
+        if self.group.reverse and self.startdelay and self.startdelayTime < self.startdelay then
+            self.startdelayTime = self.startdelayTime + elapsed
+            self.delaying = true
+        elseif not self.group.reverse and self.enddelay and self.enddelayTime < self.enddelay then
+            self.enddelayTime = self.enddelayTime + elapsed
+            self.delaying = true
+        else
+            AG:Stop(self)
+            AG:Fire(self.group, self, 'Finished')
+            return
+        end
```

Comment:
Same comment as above for this chunk.

Please preserve camelcase variable naming for field naming :)

`startdelayTime` -> `startDelayTime`

Reply:
did it globaly